linker/shared/src/test/scala/org/scalajs/linker/testutils/TestIRRepo.scala

```diff
@@ -13,22 +13,54 @@
 package org.scalajs.linker.testutils
 
 import scala.concurrent._
-import scala.concurrent.ExecutionContext.Implicits.global
 
 import org.scalajs.linker.StandardImpl
 import org.scalajs.linker.interface.IRFile
 
 object TestIRRepo {
   private val globalIRCache = StandardImpl.irFileCache()
 
-  val minilib: Future[Seq[IRFile]] = load(StdlibHolder.minilib)
-  val javalib: Future[Seq[IRFile]] = load(StdlibHolder.javalib)
+  val minilib: Future[Seq[IRFile]] = loadGlobal(StdlibHolder.minilib)
+  val javalib: Future[Seq[IRFile]] = loadGlobal(StdlibHolder.javalib)
   val empty: Future[Seq[IRFile]] = Future.successful(Nil)
-  val previousLibs: Map[String, Future[Seq[IRFile]]] =
-    StdlibHolder.previousLibs.map(x => x._1 -> load(x._2))
 
-  private def load(stdlibPath: String) = {
+  private def loadGlobal(stdlibPath: String): Future[Seq[IRFile]] = {
+    import scala.concurrent.ExecutionContext.Implicits.global
+
     Platform.loadJar(stdlibPath)
       .flatMap(globalIRCache.newCache.cached _)
   }
+
+  /** For each previous lib, calls `f(version, irFiles)`, and combines the result.
+   *
+   *  This method applies `f` *sequentially*. It waits until the returned
+   *  `Future` completes before moving on to the next iteration.
+   */
+  def sequentiallyForEachPreviousLib[A](f: (String, Seq[IRFile]) => Future[A])(
```

Comment:
Seems a bit odd to me, that the sequential processing is in here.

I can see two options:
- Just expose the bare path.
- Instead of exposing `Future[Seq[IRFile]]`, expose `() => Future[Seq[IRFile]]`.

Reply:
The `() => Future[Seq[IRFile]]` was my first idea, but it did not allow to properly `free()` the caches. Exposing the paths would make that possible, although it would then be on the caller to make sure that this done safely. If we write any new test that iterates over previous versions, we'll have to duplicate that logic.

Reply:
Ah, right. I did not take caching into account.